support-metrics-common/src/main/java/io/confluent/support/metrics/common/kafka/KafkaUtilities.java

```diff
@@ -265,8 +264,7 @@ public VerifyTopicState verifySupportTopic(ZkUtils zkUtils,
   }
 
   public boolean isReadyForMetricsCollection(KafkaServer server) {
-    return server.brokerState().currentState() == RunningAsBroker.state() ||
-        server.brokerState().currentState() == RunningAsController.state();
+    return server.brokerState().currentState() == RunningAsBroker.state();
```

Comment:
The `RunningAsController` state was only active for a short period of time even if the broker was the controller. That's why it was removed, it was not really useful. One can use `KafkaController.isActive` to know if a broker is the controller. The question about creating topics seems to be orthogonal.